### ISSUE TITLE
perfdata files are empty since 2.9

### ISSUE THREAD

**Issue by TM08353** [Icinga/icinga2#6493]
Since the upgrade to 2.9.0 no perfdata is logged and the host-perfdata.* and service-perfdata.* files are all empty. 

This all worked flawlessly in 2.8.4.

The perfdata feature is enabled, and debug log shows perfdata is written to IDO db.
(slightly sanitized statement for privacy)
```
[2018-07-27 11:46:07 +0200] debug/IdoMysqlConnection: Query: UPDATE icinga_servicestatus SET acknowledgement_type = '0',  active_checks_enabled = '1',  check_command = 'check_nrpe',  check_source = 'icinga.domain.example',  check_timeperiod_object_id = 57,  check_type = '0',  current_check_attempt = '1',  current_notification_number = '0',  current_state = '0',  endpoint_object_id = 1,  event_handler_enabled = '1',  execution_time = '0.802200',  flap_detection_enabled = '0',  has_been_checked = '1',  instance_id = 1,  is_flapping = '0',  is_reachable = '1',  last_check = FROM_UNIXTIME(1532684767),  last_hard_state = '0',  last_hard_state_change = FROM_UNIXTIME(1532658997),  last_notification = FROM_UNIXTIME(1529337096),  last_state_change = FROM_UNIXTIME(1532658997),  last_time_critical = FROM_UNIXTIME(1531126314),  last_time_ok = FROM_UNIXTIME(1532684767),  last_time_unknown = FROM_UNIXTIME(1532658939),  last_time_warning = FROM_UNIXTIME(1524735253),  latency = '0.000675',  long_output = '',  max_check_attempts = '5',  next_check = FROM_UNIXTIME(1532684826),  next_notification = FROM_UNIXTIME(1532685497),  normal_check_interval = '1',  notifications_enabled = '1',  original_attributes = 'null',  output = 'OK: ACTUAL SERVICE MESSAGE',  passive_checks_enabled = '1',  percent_state_change = '0',  perfdata = 'dp1=1145688448;3435973836;3865470566;0;4294967296 dp2=393219336;0;0;0;-1 dp3=215497288;;;0;-1 \'complex name\'=3145728;;;0;-1',  problem_has_been_acknowledged = '0',  process_performance_data = '1',  retry_check_interval = '1',  scheduled_downtime_depth = '0',  service_object_id = 1491,  should_be_scheduled = '1',  state_type = '1',  status_update_time = FROM_UNIXTIME(1532684767) WHERE service_object_id = 1491
```

perfdata files are written, but are always zero bytes
```
ls -al /var/spool/icinga2/perfdata/
total 8
drwxrwx--- 2 nagios nagios 4096 Jul 27 12:03 ./
drwxr-x--- 4 nagios nagios 4096 Oct 22  2015 ../
-rw-r--r-- 1 nagios nagios    0 Jul 27 12:03 host-perfdata.1532685814
-rw-r--r-- 1 nagios nagios    0 Jul 27 12:03 service-perfdata.1532685814
```

PerfDataWriter Object is empty as per default
```
# cat /etc/icinga2/features-enabled/perfdata.conf
/**
 * The PerfdataWriter type writes performance data files and rotates
 * them in a regular interval.
 */
object PerfdataWriter "perfdata" { }
```

## Expected Behavior
Perfdata files should contain data

## Current Behavior
files are written but always empty

## Context
pnp4nagios stopped adding data to the graphs since the 2.9 update

## Your Environment
* Version used (`icinga2 --version`):
```
icinga2 - The Icinga 2 network monitoring daemon (version: r2.9.1-1)

Copyright (c) 2012-2018 Icinga Development Team (https://www.icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Application information:
  Installation root: /usr
  Sysconf directory: /etc
  Run directory: /run
  Local state directory: /var
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid

System information:
  Platform: Ubuntu
  Platform version: 14.04.5 LTS, Trusty Tahr
  Kernel: Linux
  Kernel version: 3.13.0-153-generic
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.2
  Build host: 887a6890ccb4
```

* Operating System and version:
NAME="Ubuntu"
VERSION="14.04.5 LTS, Trusty Tahr"

* Enabled features (`icinga2 feature list`):
Disabled features: api compatlog elasticsearch gelf graphite icingastatus influxdb livestatus opentsdb statusdata syslog
Enabled features: checker command debuglog ido-mysql mainlog notification perfdata

* Config validation (`icinga2 daemon -C`):
```
icinga2 daemon -C
[2018-07-27 12:13:45 +0200] information/cli: Icinga application loader (version: r2.9.1-1)
[2018-07-27 12:13:45 +0200] information/cli: Loading configuration file(s).
[2018-07-27 12:13:47 +0200] information/ConfigItem: Committing config item(s).
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 458 Services.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 3 ServiceGroups.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 ScheduledDowntime.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 6 HostGroups.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 EventCommand.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 2 FileLoggers.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 NotificationComponent.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 5 NotificationCommands.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 513 Notifications.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 IcingaApplication.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 54 Hosts.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 Downtime.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 10 Dependencies.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 PerfdataWriter.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 6 Comments.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 CheckerComponent.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 3 Zones.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 ExternalCommandListener.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 Endpoint.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 UserGroup.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 1 IdoMysqlConnection.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 259 CheckCommands.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 7 TimePeriods.
[2018-07-27 12:13:47 +0200] information/ConfigItem: Instantiated 2 Users.
[2018-07-27 12:13:47 +0200] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2018-07-27 12:13:47 +0200] information/cli: Finished validating the configuration file(s).
```

Please let me know if this is due to our configuration (though it was working before) or a bug.

Thanks,
Tobias

**Comment by dnsmichi** [comment]
I'm not able to reproduce this here, locally on macOS.

```
mbmif /usr/local/icinga2/var/spool/icinga2 (master *) # ls -lhtR tmp perfdata/
tmp:
total 0
-rw-r--r--  1 icinga  icinga     0B Jul 27 15:50 host-perfdata
-rw-r--r--  1 icinga  icinga     0B Jul 27 15:50 service-perfdata

perfdata/:
total 440
-rw-r--r--  1 icinga  icinga    12K Jul 27 15:50 host-perfdata.1532699425
-rw-r--r--  1 icinga  icinga    49K Jul 27 15:50 service-perfdata.1532699425
-rw-r--r--  1 icinga  icinga    14K Jul 27 15:49 host-perfdata.1532699395
-rw-r--r--  1 icinga  icinga    51K Jul 27 15:49 service-perfdata.1532699395
-rw-r--r--  1 icinga  icinga    15K Jul 27 15:49 host-perfdata.1532699365
-rw-r--r--  1 icinga  icinga    50K Jul 27 15:49 service-perfdata.1532699365
-rw-r--r--  1 icinga  icinga    16K Jun 17  2016 service-perfdata.1532699335
-rw-r--r--  1 icinga  icinga     0B Jun 17  2016 host-perfdata.1532699335
```

**Comment by TM08353** [comment]
Any pointers on where to look / how to debug further?

Perfdata is reported by our plugins (and wasn't changed recently) and does show up in icingaweb2.

Is anything besides perfdata feature / PerfDataWriter needed for this to work?
Should this show up in the debug log?

**Comment by TM08353** [comment]
I did a restore of our icinga host as of last week, and re-applied the updates. 
Strangely, everything is working now.

I did a diff -r on the old and the new config (/etc/icinga2), but could not see any difference except the debuglog being enabled. 
Also, there were two backup files not ending in .conf, but to my understanding they would (or should?) have been ignored.

Confusing, but works for me now.
Thanks for looking.
